frontend/src/components/QueryBuilderV2/utils.test.ts

```diff
@@ -0,0 +1,387 @@
+/* eslint-disable sonarjs/no-duplicate-string */
+import { negateOperator, OPERATORS } from 'constants/antlrQueryConstants';
+import { extractQueryPairs } from 'utils/queryContextUtils';
+
+// Now import the function after all mocks are set up
+import { convertFiltersToExpressionWithExistingQuery } from './utils';
+
+jest.mock('utils/queryContextUtils', () => ({
+	extractQueryPairs: jest.fn(),
+}));
+
+// Type the mocked functions
+const mockExtractQueryPairs = extractQueryPairs as jest.MockedFunction<
+	typeof extractQueryPairs
+>;
+
+describe('convertFiltersToExpressionWithExistingQuery', () => {
+	beforeEach(() => {
+		jest.clearAllMocks();
+	});
+
+	test('should return filters with new expression when no existing query', () => {
+		const filters = {
+			items: [
+				{
+					id: '1',
+					key: { id: 'service.name', key: 'service.name', type: 'string' },
+					op: OPERATORS['='],
+					value: 'test-service',
+				},
+			],
+			op: 'AND',
+		};
+
+		const result = convertFiltersToExpressionWithExistingQuery(
```

Comment:
This depends on `convertFiltersToExpression`, which has an incorrect implementation https://github.com/SigNoz/signoz/pull/8804#issuecomment-3194650213